The signature of `search_function` was changed to match the new
`gtk_source_search_context_forward2()` and `gtk_source_search_context_backward2`
functions, so we have to use them instead of the now deprecated variants
without the `2` postfix.
--- a/src/ui/gui/psppire-syntax-window.c	2024-09-29 12:40:06.799137074 +0200
+++ b/src/ui/gui/psppire-syntax-window.c	2024-09-29 12:46:43.528476103 +0200
@@ -258,8 +258,8 @@ static void on_find (PsppireSyntaxWindow
                                               gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (dialog->wrap)));
 
   search_function *func = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (dialog->forward))
-                                                        ? gtk_source_search_context_forward
-                                                        : gtk_source_search_context_backward;
+                                                        ? gtk_source_search_context_forward2
+                                                        : gtk_source_search_context_backward2;
 
   gtk_text_buffer_get_iter_at_mark (buffer, &begin, mark);
   gtk_text_iter_forward_char (&begin);
